lib/graphs/digraph.nit

```diff
@@ -0,0 +1,561 @@
+# Implementations for representing directed graphs, also called digraphs.
+#
+# Currently, there are two classes
+module digraph
+
+# An arc of a digraph
+class Arc[V]
+	# The source of the arc
+	#
+	# ~~~
+	# var arc = new Arc[Int](0, 1)
+	# assert arc.source == 0
+	# ~~~
+	var source: V
+
+	# The target of the arc
+	#
+	# ~~~
+	# var arc = new Arc[Int](0, 1)
+	# assert arc.target == 1
+	# ~~~
+	var target: V
+
+	redef fun to_s: String do return "({source.to_s}, {target.to_s})"
+end
+
+# Interface for digraphs
```

Comment:
For the less mathsavvy of us, maybe you can add a little explanation of what is a digraphs and why it is usefull.